llvm/lib/Transforms/IPO/SampleProfileProbe.cpp

```diff
@@ -173,21 +173,98 @@ SampleProfileProber::SampleProfileProber(Function &Func,
   BlockProbeIds.clear();
   CallProbeIds.clear();
   LastProbeId = (uint32_t)PseudoProbeReservedId::Last;
-  computeProbeIdForBlocks();
-  computeProbeIdForCallsites();
-  computeCFGHash();
+
+  DenseSet<BasicBlock *> BlocksToIgnoreProbe;
+  DenseSet<BasicBlock *> BlocksToIgnoreCall;
```

Comment:
What's the main reason to split these into two containers? Is that because for "NewSplit" blocks, we want to ignore block probe but we still want to have call probe for calls from those blocks? 

If that's the case, suggest following renaming:

BlocksToIgnoreCall -> BlocksAndCallsToIgnore 
BlocksToIgnoreProbe -> BlocksToIgnore